platforms/android/library/src/main/java/io/element/android/wysiwyg/EditorEditText.kt

```diff
@@ -252,8 +253,46 @@ class EditorEditText : TextInputEditText {
         setSelectionFromComposerUpdate(result.selection.last)
     }
 
-    fun setLink(link: String) {
-        val result = viewModel.processInput(EditorInputAction.SetLink(link)) ?: return
+    /**
+     * Get the action that can be performed based on the current link
+     * and selection state.
+     *
+     * Based on this the caller can decide whether to call [setLink],
+     * [removeLink], or [insertLink].
+     *
+     * @return The link action or null if no action is available.
+     */
+    fun getLinkAction(): LinkAction? =
+        viewModel.getLinkAction()
+
+    /**
+     * Set a link for the current selection. This method does nothing if there is no text selected.
+     *
+     * @param link The link to set or null to remove
+     */
+    fun setLink(link: String?) {
+        val result = viewModel.processInput(
+            if(link != null) EditorInputAction.SetLink(link) else EditorInputAction.RemoveLink
+        ) ?: return
+
+        setTextFromComposerUpdate(result)
+        setSelectionFromComposerUpdate(result.selection.last)
+    }
+
+    /**
+     * Remove a link for the current selection. Convenience for setLink(null).
+     *
+     * @see [setLink]
+     */
+    fun removeLink() = setLink(null)
+
+    /**
+     * Create text with a link.
+     *
+     * @param link The link to set (or null to remove).
+     */
+    fun insertLink(link: String, text: String) {
```

Comment:
On the naming, I thought `insertLink()` could make a bit more sense for the public API than `setLinkWithText()` (to make it clearer that this will insert new text). @Velin92 what do you think? And should we consider it for the internal interfaces too?

Reply:
on the iOS side I actually named it `createLink(urlString: String, text: String)`
Since the setLink is also usable for both creating a link from scratch and for the edit, while the first is pretty much just for creation/insertion, so I guess is okay!